core/src/avm2/globals/reg_exp.rs

```diff
@@ -1,25 +1,17 @@
 //! `RegExp` impl
 
-use crate::avm2::class::Class;
 use crate::avm2::error::type_error;
-use crate::avm2::method::{Method, NativeMethodImpl, ParamConfig};
-use crate::avm2::object::{regexp_allocator, ArrayObject, FunctionObject, Object, TObject};
+use crate::avm2::object::{ArrayObject, Object, TObject};
 use crate::avm2::regexp::RegExpFlags;
 use crate::avm2::value::Value;
 use crate::avm2::Error;
-use crate::avm2::Multiname;
-use crate::avm2::QName;
 use crate::avm2::{activation::Activation, array::ArrayStorage};
 use crate::string::{AvmString, WString};
-use gc_arena::GcCell;
 
-// All of these methods will be defined as both
-// AS3 instance methods and methods on the `Array` class prototype.
-const PUBLIC_INSTANCE_AND_PROTO_METHODS: &[(&str, NativeMethodImpl)] =
-    &[("exec", exec), ("test", test)];
+pub use crate::avm2::object::reg_exp_allocator;
 
-/// Implements `RegExp`'s instance initializer.
-pub fn instance_init<'gc>(
+/// Implements `RegExp`'s `init` method, which is called from the constructor
+pub fn init<'gc>(
     activation: &mut Activation<'_, 'gc>,
     this: Option<Object<'gc>>,
     args: &[Value<'gc>],
```

Comment:
Is the `super_init` in this native constructor necessary anymore?

Reply:
Good catch